scripts/check-components

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/ruby
+# coding: utf-8
+
+require 'find'
+
+COMMON_EVENTS = ['EventsReceived', 'EventsSent']
+SOURCE_EVENTS = ['BytesReceived'] + COMMON_EVENTS
+SINK_EVENTS = COMMON_EVENTS + ['BytesSent']
+# Transforms implicitly handle the common events in the topology, and
+# have no other emit requirements.
+TRANSFORM_EVENTS = []
+
+$error_count = 0
```

Comment:
You don't need the `$` qualifier unless you want the variable to global to the entire Ruby environment. Removing it will ensure that variable is local to the script.

Reply:
The variable is used in a class method, which I understood requires the use of the qualifier. Is there another way?